airflow/Dockerfile

```diff
@@ -20,3 +13,52 @@ LABEL name="Apache Airflow" \
       description="This image is deployed by the Stackable Operator for Apache Airflow."
 
 COPY airflow/licenses /licenses
+COPY airflow/constraints-${PRODUCT}-python${PYTHON}.txt /tmp/constraints.txt
+
+# the mysql provider is currently excluded.
+# Requires implementation of https://github.com/apache/airflow/blob/2.2.5/scripts/docker/install_mysql.sh
+ENV AIRFLOW_EXTRAS=async,amazon,celery,cncf.kubernetes,docker,dask,elasticsearch,ftp,grpc,hashicorp,http,ldap,google,google_auth,microsoft.azure,odbc,pandas,postgres,redis,sendgrid,sftp,slack,ssh,statsd,virtualenv
+
+RUN microdnf update \
+        && microdnf install -y \
+            gcc \
```

Comment:
The dependencies required to build libraries should not be included in the final image. We should use a multi-staged Docker image similar to how we do it for operator images with the rust toolchain. (https://docs.docker.com/develop/develop-images/multistage-build/)

Reply:
To make the transfer of compiled libs/packages/etc to the final stage a bit less painful, we should probably use a virtual env or `--user` when calling `pip install`.

Reply:
The image now uses a multi-stage build with the virtual-env being copied across.